### PR TITLE
Adding role based access control to marketplace assistant

### PR DESCRIPTION
**Pull request by BiyonFernando**
- Upload visibility controlled APIs to vector-store with visibility roles.
- Include user roles to the prompt query for marketplace assistant.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/MarketplaceAssistantApiPublisherNotifier.java [hunk 1]
```diff
@@ -204,6 +189,13 @@ public void run() {
                 payload.put(APIConstants.API_SPEC_NAME, api.getId().getApiName());
                 payload.put(APIConstants.TENANT_DOMAIN, apiEvent.getTenantDomain());
                 payload.put(APIConstants.VERSION, apiEvent.getApiVersion());
+                payload.put(APIConstants.VISIBILITY, apiEvent.getApiVisibility());
```
Comment (ashera96) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/MarketplaceAssistantApiPublisherNotifier.java:192]: Can we not provide the role based access control using only VISIBILITYROLES? What is the need to have this VISIBILITY value?

Comment (BiyonFernando) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/MarketplaceAssistantApiPublisherNotifier.java:192]: Currently we are only using VISIBILITYROLES for filtering.

Comment (BiyonFernando) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/MarketplaceAssistantApiPublisherNotifier.java]: We removed VISIBILITY

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/MarketplaceAssistantApiPublisherNotifier.java [hunk 2]
```diff
@@ -33,6 +33,8 @@
 import org.wso2.carbon.context.CarbonContext;
 import org.wso2.carbon.apimgt.impl.utils.APIUtil;
 
+import static org.wso2.carbon.apimgt.impl.utils.APIUtil.getAPIMVersion;
```
Comment (ashera96) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/MarketplaceAssistantApiPublisherNotifier.java:36]: Why do we need a static import here? Note that `org.wso2.carbon.apimgt.impl.utils.APIUtil` is already imported in this file.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/MarketplaceAssistantApiServiceImpl.java [hunk 3]
```diff
@@ -40,11 +40,17 @@
...
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.oauth.OAuthUtil;
+import org.wso2.carbon.identity.oauth2.authz.OAuthAuthzReqMessageContext;
 
 import java.io.IOException;
 
 import javax.ws.rs.core.Response;
 
+import static org.wso2.carbon.apimgt.impl.utils.APIUtil.getAPIMVersion;
```
Comment (ashera96) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/MarketplaceAssistantApiServiceImpl.java:52]: Check this import as well.

**Review by RakhithaRR — APPROVED** [review]

**Pull request merged**